File: client/python/modcdp/NoopBrowserLauncher.py
from __future__ import annotations

from .BrowserLauncher import BrowserLaunchOptions, BrowserLauncher, LaunchedBrowser


class NoopBrowserLauncher(BrowserLauncher):
    def launch(self, options: BrowserLaunchOptions | None = None) -> LaunchedBrowser:
        self.launched = {"cdp_url": None, "ws_url": None, "close": lambda: None}
        return self.launched

File: bridge/NoopBrowserLauncher.ts
import { BrowserLauncher, type BrowserLaunchOptions, type LaunchedBrowser } from "./BrowserLauncher.js";

export class NoopBrowserLauncher extends BrowserLauncher {
  async launch(_options: BrowserLaunchOptions = {}): Promise<LaunchedBrowser> {
    this.launched = { cdp_url: null, ws_url: null, close: async () => {} };
    return this.launched;
  }
}
